src/core/thread/mle.cpp

```diff
@@ -327,7 +327,13 @@ void Mle::Restore(void)
     Get<KeyManager>().SetCurrentKeySequence(networkInfo.GetKeySequence());
     Get<KeyManager>().SetMleFrameCounter(networkInfo.GetMleFrameCounter());
     Get<KeyManager>().SetAllMacFrameCounters(networkInfo.GetMacFrameCounter());
+
+#if OPENTHREAD_MTD
+    VerifyOrExit(!IsActiveRouter(networkInfo.GetRloc16()));
```

Comment:
I am thinking whether we want to exit in this case or just continue.

For example, I think it may be good to still set the `mDeviceMode` from what we read in setting (clearning the `kModeFullThreadDevice` but keeping the rest).

The other thing I notice is that we do `Get<Mac::Mac>().SetExtAddress(networkInfo.GetExtAddress());` few lines below and I think may be good to retain the last ext address as well. 

Thoughts?

Reply:
I think we can leave it as invalid short address?  
In this case device needs to go through full attach anyways, so will need get a new RLOC16.

Reply:
I just did a quick test with current `main` build (without this PR):
- Started `ot-cli-ftd 1` and set it as a leader.
- Exited and started `ot-cli-mtd 1` now and tried to start thread (and restore the saved settings).
  - I see the device is now trying to attach (as a child) to the same network info.
  - ` mode` is indeed incorrect -> `rdn`
  - But the rloc16 seems to be fine (set to unassigned) - I guess it is cleared before re-attach:
```
> rloc16
fffe
Done
```

Reply:
It appears the code will call `SendChildUpdateRequest()`, but it won't have a valid parent, so it immediately calls `BecomeDetached()`. So I think it should work.